src/routes/auth.ts

```diff
@@ -2,16 +2,25 @@ import express from 'express';
 
 import { makeJWTToken } from '@/auth/jwt';
 import verifyGoogle from '@/auth/social/google';
+import verifyKakao from '@/auth/social/kakao';
+import verifyApple from '@/auth/social/apple';
 
 const router = express.Router();
 
 router.post(`/login/:loginType`, async (req, res, next) => {
   try {
+    const authValue = req.headers.authorization;
     let userData;
-    if (req.params.loginType === 'google') {
-      userData = await verifyGoogle(req.body.token);
-    } else if (req.params.loginType === 'kakao') {
-      // userData = await verifyKakao(req.body.token);
+    if (authValue) {
+      if (req.params.loginType === 'google') {
+        userData = await verifyGoogle(authValue);
+      } else if (req.params.loginType === 'kakao') {
+        userData = await verifyKakao(authValue);
+      } else if (req.params.loginType === 'apple-web') {
+        userData = await verifyApple(authValue, 'web');
+      } else if (req.params.loginType === 'apple-mobile') {
+        userData = await verifyApple(authValue, 'mobile');
```

Comment:
🥝 좀 더 깔끔하게 `LoginType`을 정의하고 switch-case 문으로 정리하는 거는 어떨까? 근데 switch-case문은 오히려 좀 길어지긴 해서 취향대로 선택하면 될듯!

```suggestion
      const LOGIN_TYPE = {
        google: 'google',
        kakao: 'kakao',
        appleWeb: 'apple-web',
        appleMobile: 'apple-mobile',
      } as const;
      type LoginType = typeof LOGIN_TYPE[keyof typeof LOGIN_TYPE];
      
      switch (req.params.loginType) {
        case LOGIN_TYPE.google:
          userData = await verifyGoogle(authValue);
          break;
        case LOGIN_TYPE.kakao:
          userData = await verifyKakao(authValue);
          break;
        case LOGIN_TYPE.appleWeb:
          userData = await verifyApple(authValue, 'web');
          break;
        case LOGIN_TYPE.appleMobile:
          userData = await verifyApple(authValue, 'mobile');
          break;
      }
```

Reply:
나도 `switch`문 고려하긴 했었는데, 코드가 길어지면 가독성이 떨어질거라 생각해서 `if` 사용했단 말이지..
근데 코드 비교를 해보니깐 길어지긴 해도 가독성이 좋아져서 괜찮아보이네

Reply:
근데 21번째 줄에 `LoginType`은 여기 파일에서는 사용될 일이 없어보이는데, 나중에 필요할 때 이렇게 만들면 된다고 알려주는건가..?

Reply:
@aube-dev  `LoginType`이랑 중복되는 부분이 `src/database/entity/user.ts`에 있긴한데, 다른 점이 `apple-web`, `apple-mobile`이냐 `apple`이냐 이란 말이지.. 근데 전자가 `verifyApple` 함수 내부에서만 쓰여서 따로 필요한 상황 없으면 `LoginType`은 추가하지 않아도괜찮을 것 같다고 생각하는데, 어떻게 생각해??

Reply:
음... 정말 여기에서만 쓰이는 거면 그렇게 하는 게 좋을 것 같아! 좀 아쉽긴 하지만 ㅠㅠ

Reply:
오케이~ 근데 객체의 프로퍼티 값들로 타입 만들 때 저렇게 하는거구나.. 하나 또 배웠다!!ㅋㅋ
객체의 키로 타입 만들 땐 `type LoginType = keyof typeof LOGIN_TYPE;` 이렇게 하는게 맞지?
(VSCode에선 이렇게하니깐 원하는대로 나오긴하는데 혹시 몰라서..ㅎ)